packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts

```diff
@@ -304,4 +307,39 @@ export class AppSyncSwiftVisitor extends AppSyncModelVisitor {
     }
     return !field.isNullable;
   }
+
+  protected generateAuthRules(model: CodeGenModel): string {
+    const authDirectives: AuthDirective[] = (model.directives.filter(d => d.name === 'auth') as any) as AuthDirective[];
+    const rules: string[] = [];
+    authDirectives.forEach(directive => {
+      directive.arguments?.rules.forEach(rule => {
+        const authRule = [];
+        switch (rule.allow) {
+          case AuthStrategy.owner:
+            authRule.push('allow: .owner');
+            authRule.push(`ownerField: "${rule.ownerField}"`);
+            authRule.push(`identityClaim: "${rule.identityClaim}"`);
+            break;
+          case AuthStrategy.groups:
+            authRule.push('allow: .groups');
+            authRule.push(`groupClaim: "${rule.groupClaim}"`);
+            if (rule.groups) {
+              authRule.push(`groups: [${rule.groups?.map(group => `"${group}"`).join(', ')}]`);
+            } else {
+              authRule.push(`groupsField: "${rule.groupField}"`);
+            }
```

Comment:
this looks good as well, 
1. `allow`
2. `groupClaim`
3. optional `groups`
4. or `groupsField`


```
allow: AuthStrategy,  // 1
ownerField: String? = nil,
identityClaim: String? = nil,
groupClaim: String? = nil, //2
groups: [String] = [], // 3
groupsField: String? = nil, // or 4
operations: [ModelOperation] = []
```